### ISSUE TITLE
Fails to save settings when shutting down Windows 11

### ISSUE THREAD

**Issue by almo2001** [danielchalmers/DesktopClock#7]
If I move the clock somewhere and set its color etc, it only remembers the setting if I close the window myself before shutting down the computer.

If I let the Windows shutdown sequence close the app, then it has default settings again next time I start it.

**Comment by danielchalmers** [comment]
Are you running the latest DesktopClock v3.0.0? Ugh I thought I got that fixed!

**Comment by almo2001** [comment]
Yup, I have 3.0.0. If there's anything I can do like send logs or something I'd be happy to.

**Comment by almo2001** [comment]
Thanks, got the beta. :)

**Comment by almo2001** [comment]
I was working for a while, but then it stopped. I'm getting random themes and the default location again. I'll keep an eye out for how consistent this is.

**Comment by danielchalmers** [comment]
> I was working for a while, but then it stopped. I'm getting random themes and the default location again. I'll keep an eye out for how consistent this is.

Wow very weird. Appreciate you looking into it.

Could be a difference between Shut Down and Restart?

**Comment by almo2001** [comment]
I'll let you know if I see a difference between those. I usually shut down though, but maybe there was an update or something that caused a restart.

**Comment by almo2001** [comment]
Ok, I have a (possibly useless) update. I've booted the computer after a shutdown about 5 times since the last message. The clock didn't save its preferences on the last time, but did the other 4.

I didn't see anything weird during any of the shutdowns like "this app is not allowing shutdown". If there's any other way I can help I'd be happy to.

**Comment by danielchalmers** [comment]
That's so weird. I very much appreciate the offer, but I honestly don't even know where to start. I'll keep this in mind though in case I think of anything. At least it's working most of the time lol

**Comment by almo2001** [comment]
Maybe save the prefs anytime the user makes a change to the settings? I don't know if that's a good way to do this or not. Most of my work in programming has been in games. :D

**Comment by HBA66** [comment]
Same problem here when using shutdown. On open the settings file is empty.

**Comment by danielchalmers** [comment]
@almo2001 @HBA66 do you have an antivirus?

**Comment by HBA66** [comment]
Off course. Bitdefender in the guise of Chili Security

**Comment by almo2001** [comment]
I have Windows 11, latest version, and have not installed any extra anti-virus software.

**Comment by HBA66** [comment]
> > Same problem here when using shutdown. On open the settings file is empty.
> 
> @HBA66
> 
> empty like it's 0 bytes?
Yes
> if the file was full will it reset to empty on Windows shutdown?
Yes
> and it saves properly when you Exit normally?
Yes
> is the file in an administrator privileged location?
It´s located in AppData/Local

After reinstalling using the msi-installer it seems to run OK. It even remembers the placement.

**Comment by danielchalmers** [comment]
Please let me know if you have any success with [v4.0.0](https://github.com/danielchalmers/DesktopClock/releases/tag/v4.0.0)!

**Comment by HBA66** [comment]
No matter the settings it just shows 2024 years ago! WTF! Are you kidding me? Some crazy religious joke?

**Comment by danielchalmers** [comment]
@HBA66

what region and language is your Windows in?

what does `CountdownTo` have in advanced settings? Try deleting that or the whole file.

**Comment by almo2001** [comment]
Got V4.0.0, and have not had it fail to save settings properly since. Thanks so much. :)

**Comment by danielchalmers** [comment]
> Got V4.0.0, and have not had it fail to save settings properly since. Thanks so much. :)

You don't know how relieved I am to hear that 😂😂😂
